telos-frontend/src/components/habit-tracker/JournalHabitTracker.jsx

```diff
@@ -1,34 +1,313 @@
+  };
+
+  const handleChangeOnStartDate = (event) => {
+    setStartDate(event.target.value);
+  };
+
+  const handleChangeOnEndDate = (event) => {
+    setEndDate(event.target.value);
+  };
+
+  const handleChangeOnEndType = (event) => {
+    setEndType(event.target.value);
+  };
+
+  const saveChange = () => {
+    setTitle(tempTitle);
+    setDes(tempDes);
+    addHobby(tempTitle, startDate, endDate, true);
+  };
+
+  return (
+    <div className="container">
+      <div className="HabitualsTitle">
+        <h2 align="middle">Habituals</h2>
+        <span className="AddIcon">
+          <IconButton onClick={() => setEditStatus(true)}>
+            <AddIcon />
+          </IconButton>
+        </span>
+      </div>
+      <div className="Habituals">
+        <ul>
+          {newhabit.map(
+            (item, index) =>
+              item.eventInToday && (
+                <div>
+                  <li>{`H${index + 1}`}</li>
+                  <p>{item.habitName}</p>
+                </div>
+              )
+          )}
+        </ul>
+      </div>
+
+      <Dialog
+        open={isEdit}
+        onClose={() => {
+          setEditStatus(false);
+        }}
+        classes={{
+          root: {
+            borderRadius: 16,
+          },
+        }}
+        aria-labelledby="form-dialog-title"
+      >
+        <Box className={`${isEdit === true ? 'muneContainer' : 'hidden'}`}>
+          <div className="menuHeader">
+            <Button
+              id="saveButton"
+              variant="contained"
+              onClick={() => {
+                saveChange();
+                setEditStatus(false);
+              }}
+            >
+              save
+            </Button>
+            <h2>New Habit</h2>
+            <Button
+              id="closeButton"
+              variant="contained"
+              onClick={() => {
+                setEditStatus(false);
+                setTempTitle(habitTitle);
+                setTempDes(habitDes);
+              }}
+            >
+              ×
+            </Button>
+          </div>
+          <div className="menuBody">
+            <div className="habitInfo">
+              <div className="habitName">
+                <InputLabel id="name">Name:</InputLabel>
+                <TextField
+                  id="standard-basic"
+                  value={tempTitle}
+                  onChange={handleChangeOnName}
+                  fullWidth
+                />
+              </div>
+              <div className="habitDetail">
+                <InputLabel id="description">Description:</InputLabel>
+                <TextField
+                  id="standard-basic"
+                  value={tempDes}
+                  onChange={handleChangeOnDes}
+                  fullWidth
+                />
+              </div>
+            </div>
+            <div className="weekInfo">
+              <InputLabel id="repeatsOn">Repeats on</InputLabel>
+              <div className="WeekSchedule">

```

Comment:
This can use the same sub-component used in the Dashboard view

Reply:
done